### PR TITLE
fix command line problems and actually show output

### PR DESCRIPTION
**Pull request by alexyz**
Fixed this problem with "wsconsume" on Windows:
[ERROR] 'C:\Program' is not recognized as an internal or external command,
[ERROR] operable program or batch file.

### CODE REVIEW

**Comment by arnold-maderthaner** [comment]
please merge this pull request as I also have the same issue in 1.2.3-final version

File: pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <artifactId>jaxws-tools-maven-plugin</artifactId>
   <packaging>maven-plugin</packaging>
 
-  <version>1.2.4-SNAPSHOT</version>
```
Comment (rsearls) [inline pom.xml:10]: Return to the 1.2.4-SNAPSHOT version of the POM.

File: pom.xml [hunk 2]
```diff
@@ -97,6 +97,7 @@
           <preBuildHookScript>setup.bsh</preBuildHookScript>
           <postBuildHookScript>verify.bsh</postBuildHookScript>
           <addTestClassPath>true</addTestClassPath>
+          <streamLogs>true</streamLogs>
```
Comment (rsearls) [inline pom.xml]: We do not want debugging information printed to the console by default.  
Comment out line 100 and add a comment the enabling this properly will print information to the console.

File: src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java [hunk 3]
```diff
@@ -69,17 +78,19 @@ private void runProviderInProcess(WSContractProviderParams params) throws Except
...
-      }
-      
-      MavenLogStreamConsumer out = new MavenLogStreamConsumer(log, Type.OUTPUT);
-      MavenLogStreamConsumer err = new MavenLogStreamConsumer(log, Type.ERROR);
-      int result = CommandLineUtils.executeCommandLine(new Commandline(commandLine), out, err);
+      getProviderCommandLine(commandList, params);
       
+      log.warn("Running command line: " + commandList);
+
+//      MavenLogStreamConsumer out = new MavenLogStreamConsumer(log, Type.OUTPUT);
```
Comment (rsearls) [inline src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java:82]: Remove commented lines 85-87

File: src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java [hunk 4]
```diff
@@ -111,23 +127,25 @@ private void runConsumerInProcess(WSContractConsumerParams params, String wsdl)
...
-      }
-      
-      MavenLogStreamConsumer out = new MavenLogStreamConsumer(log, Type.OUTPUT);
-      MavenLogStreamConsumer err = new MavenLogStreamConsumer(log, Type.ERROR);
-      int result = CommandLineUtils.executeCommandLine(new Commandline(commandLine), out, err);
+      getConsumerCommandLine(commandList, params, wsdl);
+
+      log.warn("Running command line: " + commandList);
+
+//      MavenLogStreamConsumer out = new MavenLogStreamConsumer(log, Type.OUTPUT);
```
Comment (rsearls) [inline src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java]: Remove commented lines 134-136

File: src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java [hunk 5]
```diff
@@ -216,16 +234,16 @@ private static String getConsumerCommandLine(List<String> commandList, WSContrac
...
       }
       commandList.add(wsdl);
-      StringBuilder command = new StringBuilder();
-      for (String s : commandList)
-      {
-         command.append(s);
-         command.append(" ");
-      }
-      return command.toString();
+//      StringBuilder command = new StringBuilder();
```
Comment (rsearls) [inline src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java:246]: Remove commented lines 237-243

File: src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java [hunk 6]
```diff
@@ -259,13 +277,13 @@ private static String getProviderCommandLine(List<String> commandList, WSContrac
...
       }
       commandList.add(params.getEndpointClass());
-      StringBuilder command = new StringBuilder();
-      for (String s : commandList)
-      {
-         command.append(s);
-         command.append(" ");
-      }
-      return command.toString();
+//      StringBuilder command = new StringBuilder();
```
Comment (rsearls) [inline src/main/java/org/jboss/ws/plugins/tools/WSContractDelegate.java:289]: Remove commented lines 280-286

File: pom.xml (hunk 2 above)
Comment (rsearls) [inline pom.xml]: I have created jira https://issues.redhat.com/browse/JBWS-4228 for this code change.
This allows us to better track changes over time.
Please amend your commit comment to start with "[JBWS-4228]"

**Review by rsearls — CHANGES_REQUESTED** [review]

**Comment by alexyz** [comment]
I tried to clean up this PR, but it failed in my local test case (Java 11) - building a slight variation of this project: https://github.com/dcm4che/dcm4chee-xds/tree/master/dcm4chee-xds2-infoset

After "run consumer out of process" I get various NoClassDefFoundError caused by bad classpath building in org.jboss.ws.plugins.tools.AbstractToolsMojo#getRequiredPluginDependencyPaths:

[WARNING] consumer: Caused by: java.lang.ClassNotFoundException: org.jboss.ws.api.tools.WSContractConsumer
This can be fixed by adding "jbossws-api".equals(s.getArtifactId())

[WARNING] consumer: Caused by: java.lang.ClassNotFoundException: org.jboss.logging.Messages
This doesn't have an obvious fix. org.jboss.logging:jboss-logging is a transitive and provided dependency of org.jboss.ws:jbossws-api. So I guess it's never delivered to the plugin? 

After adding this as an explicit dependency, I get this:
[WARNING] consumer: Caused by: java.lang.ClassNotFoundException: org.jboss.ws.tools.jaxws.impl.SunRIConsumerFactoryImpl
I don't even know what provides this.

I made some changed and pushed them but I think the plugin has more fundamental problems. I'm not interested in using the plugin anymore so someone else can take this over.

**Comment by rsearls** [comment]
What is your variation to https://github.com/dcm4che/dcm4chee-xds/tree/master/dcm4chee-xds2-infoset
I would like to investigate the nature of the failure.

I integrated your changes  into  a branch of jaxws-tools-maven-plugin.  It builds and runs the tests successfully.  I have submitted PR https://github.com/jbossws/jaxws-tools-maven-plugin/pull/10

**Comment by alexyz** [comment]
Sorry forgot to reply sooner.
The only real changes for the infoset project are to the pom:
Removing the parent (it's virtually independent of the rest of DCM4CHE)
Changing the WS plugin name and version to the latest

**Comment by rsearls** [comment]
Closing this PR without merging.  PR https://github.com/jbossws/jaxws-tools-maven-plugin/pull/10
appears to address the issue.

**Pull request closed without merging**